modules/stochastic_tools/src/covariances/LMC.C

```diff
@@ -158,39 +162,45 @@ LMC::computedKdhyper(RealEigenMatrix & dKdhp,
 }
 
 void
-LMC::computeBMatrix(RealEigenMatrix & Bmat, const unsigned int exp_i) const
+LMC::computeBMatrix(torch::Tensor & Bmat, const unsigned int exp_i) const
 {
+  auto Bmat_accessor = Bmat.accessor<Real, 2>();
   const auto & a_coeffs = *_a_coeffs[exp_i];
   const auto & lambda_coeffs = *_lambdas[exp_i];
 
   for (const auto row_i : make_range(_num_outputs))
     for (const auto col_i : make_range(_num_outputs))
     {
-      Bmat(row_i, col_i) = a_coeffs[row_i] * a_coeffs[col_i];
+      Bmat_accessor[row_i][col_i] = a_coeffs[row_i] * a_coeffs[col_i];
       if (row_i == col_i)
-        Bmat(row_i, col_i) += lambda_coeffs[col_i];
+        Bmat_accessor[row_i][col_i] += lambda_coeffs[col_i];
```

Comment:
this is just a*a^T+lambda*I, can be defnitely accelerated with matrix operations in torch.